Gem/Core/Services/NavigationStateManager.swift

```diff
@@ -0,0 +1,86 @@
+// Copyright (c). Gem Wallet. All rights reserved.
+
+import SwiftUI
+
+typealias TabScrollToTopId = TabItem
+
+protocol NavigationStateManagable: Observable {
+    var walletNavigationPath: NavigationPath { get set }
```

Comment:
```suggestion
    var wallet: NavigationPath { get set }
```

to simplify naming, as it's called `NavigationStateManager`